src/object/srv_obj.c

```diff
@@ -2017,9 +2028,17 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 	 * modification or not, so still need to dispatch
 	 * the RPC to other replicas.
 	 */
+
+	/* For EC object modification, ask DTX to 'sync' commit. */
+	if (orw->orw_flags & ORF_DTX_SYNC ||
+	    daos_oclass_is_ec(orw->orw_oid.id_pub, NULL))
```

Comment:
seems client already set the ORF_DTX_SYNC flags for EC, so probably need not check daos_oclass_is_ec() again?
because daos_oclass_is_ec() need to lookup oca list for the oid, with some overhead, if must check if it is EC obj, may reuse ORF_EC flag to avoid check oca again at server-side.

Reply:
Right, will remove it when refresh it next time.